crates/next-core/src/router_source.rs

```diff
@@ -91,6 +91,9 @@ impl ContentSource for NextRouterContentSource {
                 this.inner
                     .get(path, Value::new(ContentSourceData::default()))
             }
+            RouterResult::None => this
+                .inner
+                .get(path, Value::new(ContentSourceData::default())),
```

Comment:
Shouldn't this be returning a 404? Possibly with an emitted issue saying you might have forgotten basepath?

Reply:
Since the Next.js router will also return "none" for page/app sources' assets, we still need to check whether the inner source has some content for this path.